src/engine/server_iv.c

```diff
@@ -1105,16 +1117,35 @@ iv_op_async(struct ds_iv_ns *ns, struct ds_iv_key *key, d_sg_list_t *value,
 
 	memcpy(&ult_arg->iv_key, key, sizeof(*key));
 	ult_arg->shortcut = shortcut;
-	ult_arg->iv_sync = *sync;
+	if (sync != NULL)
+		ult_arg->iv_sync = *sync;
 	ult_arg->retry = retry;
 	ds_iv_ns_get(ns);
 	ult_arg->ns = ns;
 	ult_arg->opc = opc;
+
+	/* Synchronous mode, let's create the eventual to wait */
+	if (sync == NULL || sync->ivs_mode != CRT_IV_SYNC_LAZY) {
+		rc = ABT_eventual_create(sizeof(*status), &done_eventual);
+		if (rc)
+			return rc;
+		ult_arg->eventual = done_eventual;
```

Comment:
if "sync->ivs_mode == CRT_IV_SYNC_LAZY", seems no place to create ult_arg->eventual? Then in "iv_op_ult() -> ABT_eventual_set(ult_arg->eventual, NULL, rc)" the ult_arg->eventual is NULL?